website/docs/language/resources/ephemeral/index.mdx

```diff
@@ -0,0 +1,73 @@
+
+## Ephemeral resources
+
+Ephemeral resources are Terraform resources that are essentially temporary. Ephemeral resources have a unique lifecycle, and Terraform does not store them in its state. Each `ephemeral` block describes one or more ephemeral resources, such as a temporary password or connection to another system.
+
+In your configuration, you can only reference an `ephemeral` block in [other ephemeral contexts](/terraform/language/resources/ephemeral/reference#reference-ephemeral-resources).
+
+### Lifecycle
+
+The lifecycle of an ephemeral resource is different from resources and data sources. When Terraform provisions ephemeral resources, it performs the following steps:
+
+1. If Terraform needs to access the result of an ephemeral resource, it opens
+that ephemeral resource. For example, if Terraform opens an ephemeral resource for a Vault secret, the Vault provider obtains a lease and returns a secret. 
+
+1. If Terraform needs access to the ephemeral resource for longer than the
+remote system's enforced expiration time, Terraform asks the provider
+to renew it periodically. For example, if Terraform renews a Vault secret ephemeral resource, the Vault provider then calls Vault's lease renewal API endpoint to extend the expiration time.
+
+1. Once Terraform no longer needs an ephemeral resource, Terraform closes
+it. This happens after the providers that depend on a certain ephemeral resource
+complete all of their work for the current Terraform run phase. For example, closing a Vault secret ephemeral resource means the Vault provider explicitly ends the lease, allowing Vault to immediately revoke the associated credentials.
+
+Terraform follows these lifecycle steps for each instance of an ephemeral
+resource in a given configuration.
+
+### Configuration model
+
+To learn more about the `ephemeral` block, refer to the [Ephemeral resource reference](/terraform/language/resources/ephemeral/reference).
+
+## Write-only arguments
+
+-> **Public Beta**: Write-only arguments are in public beta and available in Terraform v1.11 and later.  Public beta features and APIs are subject to change.
+
+Terraform resources can include ephemeral arguments, also known as attributes, for data that only needs to exist temporarily. An ephemeral argument on a resource is called a "write-only argument". Write-only arguments can help store generated sensitive data for the current Terraform operation, such as a short-lived password, token, or session identifier.
+
+Write-only arguments are only available during runtime, and Terraform omits them from state and plan files. On a new Terraform operation, a write-only argument always start as `null` before Terraform overwrites it with a new value from your configuration.
```

Comment:
```suggestion
Write-only arguments are only available during runtime, and Terraform omits them from state and plan files. On a new Terraform operation, a write-only argument always starts as `null` before Terraform overwrites it with a new value from your configuration.
```

Re: write-only arguments starting as `null` - I'm not clear as to why this fact is relevant to a practitioner?

Reply:
@robin-norwood I think it's important so they know what the default value of write-only arguments are before they assign a value. 

Let me try a rephrase to make that clearer:
```suggestion
Write-only arguments are only available during runtime, and Terraform omits them from state and plan files. By default, write-only argument evaluate as `null` at the start of every Terraform operation.
```